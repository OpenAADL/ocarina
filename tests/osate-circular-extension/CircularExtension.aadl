-- bad: self extension
device dev1 extends dev1
end dev1;

-- good: checking of dev1 breaks the loop
device dev2 extends dev1
end dev2;

-- bad: cyclic extension between dev3 and dev4
device dev3 extends dev4
end dev3;

-- good: error reported on dev3
device dev4 extends dev3
end dev4;

system s
end s;

-- bad: cycle between s.i and s.j
system implementation s.i extends s.j
end s.i;

-- good: error reported on s.i
system implementation s.j extends s.i
end s.j;

-- bad: cycle between pgt and pgt2
port group pgt extends pgt2
  features
end pgt;

-- good: error reported on pgt
port group pgt2 extends pgt2
  features
end pgt2;

