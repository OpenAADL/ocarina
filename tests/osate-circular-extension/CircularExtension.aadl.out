CircularExtension.aadl:2:01: component type creates a circular extension 
CircularExtension.aadl:2:01: component type creates a circular extension 
CircularExtension.aadl:10:01: component type creates a circular extension 
CircularExtension.aadl:14:01: component type creates a circular extension 
CircularExtension.aadl:21:01: component implementation creates a circular extension 
CircularExtension.aadl:25:01: component implementation creates a circular extension 
CircularExtension.aadl:34:01: feature group type creates a circular extension 
CircularExtension.aadl:34:01: feature group type creates a circular extension 
Cannot analyze AADL specifications
